azext_edge/edge/providers/check/base.py

```diff
@@ -71,8 +73,14 @@ def check_post_deployment(
 
     if lowercase_api_resources:
         for resource, evaluate_func in evaluate_funcs.items():
-            if (resource == CORE_SERVICE_RUNTIME_RESOURCE) or\
-                    (resource.value in lowercase_api_resources and check_resources[resource]):
+            append_resource = False
+            # only add core service evaluation if there is no resource filter
+            if resource == CoreServiceResourceKinds.RUNTIME_RESOURCE and not resource_kinds:
+                append_resource = True
+            elif (resource and resource.value in lowercase_api_resources and check_resources[resource]):
+                append_resource = True
+
+            if append_resource:
                 result["postDeployment"].append(evaluate_func(detail_level=detail_level, as_list=as_list))
```

Comment:
For a future PR, I think we can consolidate the logic here with the `for resource in resource_kinds_enum:` loop to make it a bit easier to follow, but as long as the logic works for now I'm good with it.

Reply:
got it, I will note it down and change in the DR pr